### PR TITLE
add support for windows xp using NtQueryObject in ntdll.dll

### PR DESCRIPTION
**Pull request by mysqto**
Since GetFileInformationByHandleEx is not supported under Windows Vista, Using NtQueryObject in ntdll.dll to get full name of cygwin/msys pty pipe under Windows XP

### CODE REVIEW

File: example_test.go [hunk 1]
```diff
@@ -4,7 +4,7 @@ import (
 	"fmt"
 	"os"
 
-	"github.com/mattn/go-isatty"
+	"github.com/mysqto/isatty"
```
Comment (mattn) [inline example_test.go]: Do not change this line.

File: go.mod [hunk 2]
```diff
@@ -1,3 +1,3 @@
-module github.com/mattn/go-isatty
+module github.com/mysqto/isatty
```
Comment (mattn) [inline go.mod]: Please revert this.

File: isatty_windows.go [hunk 3]
```diff
@@ -4,22 +4,26 @@
 package isatty
 
 import (
+	"fmt"
```
Comment (mattn) [inline isatty_windows.go]: Use errors.New instead.

File: isatty_windows.go [hunk 4]
```diff
@@ -4,22 +4,26 @@
...
 	"syscall"
 	"unicode/utf16"
 	"unsafe"
 )
 
 const (
-	fileNameInfo uintptr = 2
-	fileTypePipe         = 3
+	objectNameInfo uintptr = 1
+	fileNameInfo   uintptr = 2
```
Comment (mattn) [inline isatty_windows.go]: remove uintptr of this line.

File: isatty_windows.go [hunk 5]
```diff
@@ -61,18 +68,42 @@ func isCygwinPipeName(name string) bool {
 		return false
 	}
 
-	if token[4] != "master" {
+	if token[4] != `master` {
```
Comment (mattn) [inline isatty_windows.go]: Do not use this style.

File: go.mod (hunk 2 above)
Comment (mysqto) [reply go.mod]: Sorry, I commit some personal code before this PR is merged

File: isatty_windows.go (hunk 5 above)
Comment (mysqto) [reply isatty_windows.go]: fixed

File: isatty_windows_test.go [hunk 6]
```diff
@@ -23,6 +23,9 @@ func TestCygwinPipeName(t *testing.T) {
 		{`\cygwin-e022582115c10879-pty4-from-master`, true},
 		{`\msys-e022582115c10879-pty4-to-master`, true},
 		{`\cygwin-e022582115c10879-pty4-to-master`, true},
+		{`\Device\NamedPipe\cygwin-e022582115c10879-pty4-from-master`, true},
+		{`Device\NamedPipe\msys-e022582115c10879-pty4-to-master`, true},
+		{`Device\NamedPipe\cygwin-e022582115c10879-pty4-to-master`, true},
```
Comment (mattn) [inline isatty_windows_test.go]: you change is checking prefix string `\Device\...` but this test does not have leading backslash.

File: isatty_windows.go [hunk 7]
```diff
@@ -4,22 +4,26 @@
...
-	fileNameInfo uintptr = 2
-	fileTypePipe         = 3
+	objectNameInfo uintptr = 1
+	fileNameInfo           = 2
+	fileTypePipe           = 3
 )
 
 var (
 	kernel32                         = syscall.NewLazyDLL("kernel32.dll")
+	ntdll                            = syscall.NewLazyDLL("Ntdll.dll")
```
Comment (mattn) [inline isatty_windows.go]: s/Ntdll.dll/ntdll.dll/

**Comment by mattn** [comment]
Thank you

**Pull request merged**